### DISCUSSION TITLE
Is there any way to change the aggregation temporality according to the metric name (or other label)?

### DISCUSSION THREAD

**Discussion by huange7** [open-telemetry/opentelemetry-java#6842] [Q&A]
When I set 'otel.exporter.otlp.metrics.temporality.preference=delta', the aggregation temporality is set according to the instrument type. I want to manually specify it according to the metric name or when creating a new metric. Is this allowed?

Any reply is appreciated. Thank you.

**Comment by jack-berg** [comment]
No there is not. In old old prototypes, the aggregation temporality was configurable at the instrument level, and you could use views to configure it for a single instrument. This feature was pulled before the stable release because we realized that the temporality requirement is really a property of the backend you're sending data to. I.e. backend A supports cumulative, backend B supports delta, backend C supports either. And so we adjusted the design to allow each MetricReader (and associated MetricExporter by extension) to specify the aggregation temporality they require as a function of instrument kind. 

[MetricExporter](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/MetricExporter.java#L31) extends [AggregationTemporalitySelector](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/AggregationTemporalitySelector.java#L79). To specify a aggergation temporality preference, MetricExporter implementations implement `AggregationTemporalitySelector#getAggregationTemporality(InstrumentTYpe)`.

For the generic OTLP metric exporters, the temporality selector is configurable here: https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/http/metrics/OtlpHttpMetricExporterBuilder.java#L193-L198

**Reply by huange7** [reply]
Hi, jack. So, an exporter can only support one kind of aggregation temporality, right?

**Reply by jack-berg** [reply]
One aggregation temporality for each instrument type. For example, you could select delta histograms, and cumulative for all others instrument types.
